Error (WERROR=1) while compiling the CBOR module

While testing some stuff in the mulle platform I notice that the unittests test doesn't compile anymore. This is due to the CBOR module which throws the following error (warning):

```console
/Users/facosta/git/RIOT2/RIOT/sys/cbor/cbor.c: In function 'cbor_deserialize_date_time':
/Users/facosta/git/RIOT2/RIOT/sys/cbor/cbor.c:770:9: error: implicit declaration of function 'strptime' [-Werror=implicit-function-declaration]
     if (strptime(buffer, format, val) == 0) {
         ^~~~~~~~
cc1: all warnings being treated as errors
```

I tried several methods to make it disappear without success.

Besides, the CBOR test (not unittest) has now a stack overflow probably due to #7230 but I still need to verify.

For the cbor test I get this:

```console
snake:RIOT facosta$ cd tests/cbor/
snake:cbor facosta$ BOARD=mulle make all
Building application "tests_cbor" for "mulle" with MCU "k60".

"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/boards/mulle
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/core
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/cpu/k60
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/cpu/cortexm_common
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/cpu/cortexm_common/periph
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/cpu/k60/periph
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/cpu/kinetis_common
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/cpu/kinetis_common/periph
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/drivers
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/drivers/mtd
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/drivers/mtd_spi_nor
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/drivers/nvram
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/drivers/nvram_spi
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/drivers/periph_common
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/sys
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/sys/auto_init
"/Applications/Xcode.app/Contents/Developer/usr/bin/make" -C /Users/facosta/git/RIOT2/RIOT/sys/cbor
/Users/facosta/git/RIOT2/RIOT/sys/cbor/cbor.c: In function 'cbor_deserialize_date_time':
/Users/facosta/git/RIOT2/RIOT/sys/cbor/cbor.c:769:9: error: implicit declaration of function 'strptime' [-Werror=implicit-function-declaration]
     if (strptime(buffer, format, val) == 0) {
         ^~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [/Users/facosta/git/RIOT2/RIOT/tests/cbor/bin/mulle/cbor/cbor.o] Error 1
make[2]: *** [ALL--/Users/facosta/git/RIOT2/RIOT/sys/cbor] Error 2
make[1]: *** [ALL--/Users/facosta/git/RIOT2/RIOT/sys] Error 2
make: *** [link] Error 2
snake:cbor facosta$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (GNU Tools for ARM Embedded Processors 6-2017-q2-update) 6.3.1 20170620 (release) [ARM/embedded-6-branch revision 249437]
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

For the unittests I get exactly the same.

I also got the same error with gcc 5.2 (I updated it because I thought that was the problem).

--- a/Makefile.dep
+++ b/Makefile.dep
@@ -8,6 +8,12 @@ OLD_USEPKG := $(sort $(USEPKG))
 # pull dependencies from drivers
 include $(RIOTBASE)/drivers/Makefile.dep
 
+ifneq (,$(filter cbor_ctime,$(USEMODULE)))
+  ifneq (,$(filter newlib,$(USEMODULE)))
+    USEMODULE += newlib_gnu_source
+  endif
+endif
+
 ifneq (,$(filter csma_sender,$(USEMODULE)))
   USEMODULE += random
   USEMODULE += xtimer
@@ -379,6 +385,10 @@ endif
 
 ifneq (,$(filter od,$(USEMODULE)))
   USEMODULE += fmt
+endif
+
+ifneq (,$(filter newlib_gnu_source,$(USEMODULE)))
+  USEMODULE += newlib
 endif
 
 ifneq (,$(filter newlib_nano,$(USEMODULE)))

--- a/makefiles/libc/newlib.mk
+++ b/makefiles/libc/newlib.mk
@@ -9,6 +9,10 @@ ifneq (,$(filter newlib_nano,$(USEMODULE)))
   endif
 endif
 
+ifneq (,$(filter newlib_gnu_source,$(USEMODULE)))
+  CFLAGS += -D_GNU_SOURCE=1
+endif
+
 ifeq (1,$(USE_NEWLIB_NANO))
   export LINKFLAGS += -specs=nano.specs
 endif

--- a/makefiles/pseudomodules.inc.mk
+++ b/makefiles/pseudomodules.inc.mk
@@ -52,6 +52,7 @@ PSEUDOMODULES += netstats_l2
 PSEUDOMODULES += netstats_ipv6
 PSEUDOMODULES += netstats_rpl
 PSEUDOMODULES += newlib
+PSEUDOMODULES += newlib_gnu_source
 PSEUDOMODULES += newlib_nano
 PSEUDOMODULES += openthread
 PSEUDOMODULES += pktqueue

--- a/sys/cbor/Makefile
+++ b/sys/cbor/Makefile
@@ -1,7 +1,7 @@
 MODULE = cbor
 
 ifneq ($(shell uname -s),Darwin)
-	CFLAGS += -D_XOPEN_SOURCE=600
+  CFLAGS += -D_XOPEN_SOURCE=600
 endif
 
 include $(RIOTBASE)/Makefile.base

--- a/sys/cbor/cbor.c
+++ b/sys/cbor/cbor.c
@@ -24,6 +24,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef MODULE_CBOR_CTIME
+#include <time.h>
+#endif
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 

